src/app/answer-details/answer-details.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit, Input } from "@angular/core";
+import { Answer } from "../models/answer";
+import { AnswerViewModel } from "../shared/answerViewModel";
+import { FormService } from "../form-service/form.service";
+import { ActivatedRoute } from "@angular/router";
+import { from, Observable, empty } from "rxjs";
+import { switchMap, tap } from "rxjs/operators";
+import { FormBuilder, Validators } from "@angular/forms";
+
+@Component({
+  selector: "app-answer-details",
+  templateUrl: "./answer-details.component.html",
+  styleUrls: ["./answer-details.component.css"]
+})
+export class AnswerDetailsComponent implements OnInit {
+  public answer$: Observable<AnswerViewModel>;
+  public questions: Answer[];
+
+  constructor(
+    private formService: FormService,
+    private route: ActivatedRoute,
+    private fb: FormBuilder
+  ) {}
+  valuationForm = this.fb.group({
+    valuation: [
+      "",
+      [Validators.required, Validators.min(0), Validators.max(100)]
+    ],
+    notes: ["", [Validators.required, Validators.maxLength(450)]],
+    state: ["", [Validators.required]]
+  });
+  get valuation() {
+    return this.valuationForm.get("valuation");
+  }
+  get notes() {
+    return this.valuationForm.get("notes");
+  }
+  get state() {
+    return this.valuationForm.get("state");
+  }
+
+  ngOnInit(): void {
+    this.formService.fetchQuestions().subscribe(data => {
+      this.questions = data;
+      console.log(this.questions);
+    });
+
+    this.answer$ = from(this.route.paramMap).pipe(
+      switchMap(params => {
+        return this.formService.fetchAnswer({ id: params.get("id") });
+      })
+    );
+
+    console.log("Hello");
+    this.answer$.subscribe(res => console.log(res));
+  }
+  onSubmit() {}
+
+  getFullQuestion(id: string): string {
+    for (let index = 0; index < this.questions.length; index++) {
```

Comment:
this.questions.find(question => question.id === id).fullQuestion;

Try to avoid simple for loops in JS, there are plenty of more elegant ways to cycle through arrays. Check ES6 map, filter, find, reduce, forEach methods for more info.

Reply:
This is fixed and pushed to frontend-code-fixes.